### PR TITLE
do not expose Service

### PR DESCRIPTION
**Pull request by pongad**
I'm not sure how ready this PR is. The general logic is to hide the implementation in a subclass then create a delegator. If the approach looks fine to you, I'll go ahead and document the methods.

### CODE REVIEW

**Comment by pongad** [comment]
@garrettjonesgoogle Please take a look. If possible, I'd like to merge this ASAP, since everything we do will conflict with this change.

**Comment by garrettjonesgoogle** [comment]
It's hard to figure out what has changed and what has moved - can you point me to the points in the PR that need to actually be considered?

**Comment by pongad** [comment]
@garrettjonesgoogle Essentially, all fields (non static ones) and all methods are moved into the inner `SubscriberImpl` class. `SubscriberImpl` still `extends AbstractService` so that we can the methods we want. The `Subscriber` class just re-declare methods in `SubscriberImpl`. It also re-declares `Service.State` and `Service.Listener` classes and translate between the original and the redeclaration.

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java [hunk 1]
```diff
@@ -115,272 +117,402 @@
...
+    impl.stopAsync();
+    return this;
   }
 
-  @Override
-  protected void doStart() {
-    logger.debug("Starting subscriber group.");
-    startStreamingConnections();
-    notifyStarted();
+  public enum State {
```
Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java]: Redeclaring `Service.State`.

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java [hunk 2]
```diff
@@ -115,272 +117,402 @@
...
+    FAILED,
+    NEW,
+    RUNNING,
+    STARTING,
+    STOPPING,
+    TERMINATED
+  }
+
+  public abstract static class SubscriberListener {
```
Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Subscriber.java]: Redeclare `Service.Listener`.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**